apps/heft/src/plugins/CopyFilesPlugin.ts

```diff
@@ -0,0 +1,336 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as chokidar from 'chokidar';
+import * as path from 'path';
+import glob from 'fast-glob';
+import { performance } from 'perf_hooks';
+import { AlreadyExistsBehavior, FileSystem } from '@rushstack/node-core-library';
+import { TapOptions } from 'tapable';
+
+import { IHeftPlugin } from '../pluginFramework/IHeftPlugin';
+import { HeftSession } from '../pluginFramework/HeftSession';
+import { HeftConfiguration } from '../configuration/HeftConfiguration';
+import { ScopedLogger } from '../pluginFramework/logging/ScopedLogger';
+import { Async } from '../utilities/Async';
+import {
+  IHeftEventActions,
+  CoreConfigFiles,
+  HeftEvent,
+  IExtendedSharedCopyConfiguration
+} from '../utilities/CoreConfigFiles';
+import {
+  IBuildStageContext,
+  IBundleSubstage,
+  ICompileSubstage,
+  IPostBuildSubstage,
+  IPreCompileSubstage
+} from '../stages/BuildStage';
+import { Constants } from '../utilities/Constants';
+
+const PLUGIN_NAME: string = 'CopyFilesPlugin';
+const HEFT_STAGE_TAP: TapOptions<'promise'> = {
+  name: PLUGIN_NAME,
+  stage: Number.MAX_SAFE_INTEGER / 2 // This should give us some certainty that this will run after other plugins
+};
+
+interface ICopyFileDescriptor {
+  sourceFilePath: string;
+  destinationFilePaths: string[];
```

Comment:
Would it be much work to create a test project that illustrates some common cases:
- Copying one file
- Copying one file to multiple target folders
- Copying several files in a nested folder tree
- Copying a file into a folder that does not already exist

Reply:
Added a test project.